src/Reflection/ClassReflection.php

```diff
@@ -113,65 +108,55 @@ public function getMethod($name)
     }
 
     /**
-     * Get reflection objects of all methods
+     * {@inheritDoc}
      *
      * @param  int $filter
-     * @return MethodReflection[]
+     * @return list<MethodReflection>
      */
     #[ReturnTypeWillChange]
     public function getMethods($filter = -1)
     {
-        $methods = [];
-        foreach (parent::getMethods($filter) as $method) {
-            $instance  = new MethodReflection($this->getName(), $method->getName());
-            $methods[] = $instance;
-        }
+        $name = $this->getName();
 
-        return $methods;
+        return array_map(
+            static fn (ReflectionMethod $method): MethodReflection => new MethodReflection($name, $method->getName()),
+            parent::getMethods($filter)
+        );
     }
 
     /**
-     * Returns an array of reflection classes of traits used by this class.
+     * {@inheritDoc}
      *
-     * @return null|array
+     * @return array<trait-string, ClassReflection>
      */
     #[ReturnTypeWillChange]
     public function getTraits()
     {
-        $vals   = [];
-        $traits = parent::getTraits();
-        if ($traits === null) {
-            return;
-        }
-
-        foreach ($traits as $trait) {
-            $vals[] = new ClassReflection($trait->getName());
-        }
-
-        return $vals;
+        return array_map(
+            static fn (ReflectionClass $trait): ClassReflection => new ClassReflection($trait->getName()),
+            parent::getTraits()
+        );
     }
 
     /**
-     * Get parent reflection class of reflected class
+     * {@inheritDoc}
      *
-     * @return ClassReflection|bool
+     * @return ClassReflection|false
      */
     #[ReturnTypeWillChange]
     public function getParentClass()
     {
-        $phpReflection = parent::getParentClass();
-        if ($phpReflection) {
-            $laminasReflection = new ClassReflection($phpReflection->getName());
-            unset($phpReflection);
+        $reflection = parent::getParentClass();
 
-            return $laminasReflection;
+        if (! $reflection) {
+            return false;
         }
 
-        return false;
+        return new ClassReflection($reflection->getName());
     }
 
     /**
-     * Return reflection property of this class by name
+     * {@inheritDoc}
```

Comment:
Since this should be importing the whole docBlock, I would make it explicit https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc-tags.md#41-making-inheritance-explicit-using-the-inheritdoc-tag.

Reply:
dd68158de612598d53a7f3dcc96c27c5b3816dbf